Error: Difícil ingresar múltiples números negativos en operaciones

### 🐞 Descripción del error

La calculadora GUI presenta dos problemas principales detectados durante pruebas manuales:

1.  **Dificultad para ingresar múltiples números negativos en funciones científicas como min(-5, -10) o max(-3, -8)**
   - Para ingresar un número negativo se debe hacer una resta (ej: `0 - 5 =`).  Pero al intentar usar ese resultado en funciones como `min`/`max` y luego ingresar otro número negativo, la lógica se complica y produce resultados incorrectos.
   - **Ejemplo:** Al intentar calcular `min(-5, -10)`, el resultado obtenido es incorrecto (`-15.0` en vez de `-10.0`).

2. **Problema al operar cuando el primer número es 0**
   - Si el primer número ingresado es `0` y luego se realiza cualquier operación con otro número, la calculadora no hace nada, como si el `0` estuviera vacío o nulo.  La operación no se ejecuta y no aparece ningún resultado en pantalla.
   - **Ejemplo:** Ingresar `0`, presionar `-`, luego `5`, y luego `=`.  No muestra ningún resultado.

**Detectado durante:** Testing del issue #23

---

### 🔁 Pasos para reproducir

#### Para el problema de números negativos:
1. Abrir la calculadora GUI: `python gui.py`
2.  Intentar calcular `min(-5, -10)`: 
   - Presionar: `0`, `-`, `5`, `=` → Display: `-5` ✅
   - Presionar: `min` → Prepara para segundo número
   - Presionar: `0`, `-`, `1`, `0`, `=` → Calcula `0 - 10 = -10`
   - Presionar: `=` nuevamente para calcular min
3. Observar el resultado incorrecto
   - **Resultado obtenido:** `-15.0` ❌ (incorrecto)
   - **Resultado esperado:** `-10.0` ✅

#### Para el problema del primer número cero:
1.  Abrir la calculadora GUI: `python gui.py`
2. Ingresar `0`, luego presionar cualquier operador (`+`, `-`, etc.) y luego ingresar otro número (`5` por ejemplo), y luego `=`.
3. La calculadora no realiza ninguna operación ni muestra resultado.

---

### 📷 Evidencia (opcional)

**Test que falló:**
```python
def test_valor_minimo_negativos():
    """Test de min con números negativos: min(-5, -10) = -10"""
    calc.number_button_click('0')
    calc.operation_click('-')
    calc.number_button_click('5')
    calc.equals_click()
    calc.scientific_click('min')
    calc.number_button_click('0')
    calc.operation_click('-')
    calc.number_button_click('1')
    calc.number_button_click('0')
    calc. equals_click()
    calc.equals_click()
    # Falla: AssertionError: assert -15.0 == -10.0
```

**Test para el primer número 0:**
```python
def test_operacion_con_cero():
    """Test ingreso 0 seguido de operación debería funcionar correctamente: 0 - 5 = -5"""
    calc. number_button_click('0')
    calc.operation_click('-')
    calc.number_button_click('5')
    calc.equals_click()
    assert calc.current_value == '-5'
    # Actualmente no muestra ningún resultado
```

**Log del error:**
```
E       AssertionError: assert -15.0 == -10.0
E       AssertionError: assert '' == '-5' (para el bug con 0)
```

---

### ✅ Comportamiento esperado

**Para números negativos:**
- Debería poder ingresar y operar múltiple números negativos en funciones como min/max (por ejemplo, min(-5, -10) = -10).
- Botón `+/-` para cambiar signo sería lo ideal; o soporte de la tecla menos al inicio.

**Para el primer número 0:**
- La calculadora debe permitir usar `0` como número válido y realizar operaciones, mostrándolas correctamente en pantalla.
- Por ejemplo, `0 - 5 = -5`, `0 + 8 = 8`, etc.

---

### 💡 Solución propuesta

1. Implementar botón `+/-` en la GUI o permitir la tecla menos al inicio para ingresar números negativos de manera directa y clara.
2. Corregir la lógica para que el número `0` se trate como un número válido y no como vacío, permitiendo realizar operaciones correctamente cuando el primer número es cero.

---

### 🔗 Referencias
- **Issue relacionado:** #23 (Testing de GUI)
- **Archivo afectado:** `gui.py`
- **Test que detectó los bugs:** `test_gui_calculator.py::test_valor_minimo_negativos`, `test_gui_calculator.py::test_operacion_con_cero`

---

### 📊 Severidad

**Media** - No impide el uso básico de la calculadora, pero limita funcionalidad con números negativos y operaciones que inicien con cero.

---

### 👥 Asignación

Se puede asignar a quien esté trabajando en mejoras de la GUI o dejarlo para trabajo futuro.

--- a/README.md
+++ b/README.md
@@ -19,9 +19,12 @@ Este repositorio está diseñado para practicar y aprender el flujo de trabajo c
 - Git
 - Editor de código (VS Code recomendado)
 
-### Próximas Características
+### Características Implementadas
 
 - Interfaz gráfica con tkinter
+
+### Próximas Características
+
 - Más operaciones matemáticas
 - Historial de operaciones
 - Soporte para expresiones matemáticas complejas
@@ -43,9 +46,10 @@ pip install pytest
 
 ```
 team-practice/
-├── calculator.py        # Código fuente principal
-├── test_calculator.py   # Pruebas unitarias
+├── calculator.py       # Código fuente principal
+├── test_calculator.py  # Pruebas unitarias
 ├── main.py             # Programa principal
+├── gui.py              # Interfaz gráfica      
 ├── .gitignore          # Archivos ignorados por Git
 ├── .github/            # Configuración y templates de GitHub
 │   ├── ISSUE_TEMPLATE/ # Plantillas para issues
@@ -54,7 +58,7 @@ team-practice/
 │   └── REVIEW_COMMENTS.md      # Plantillas para comentarios de revisión
 ├── README.md           # Este archivo
 ├── CHANGELOG.md        # Historial de cambios
-└── LICENSE            # Licencia del proyecto
+└── LICENSE             # Licencia del proyecto
 ```
 
 ## 🤝 Flujo de Trabajo
@@ -112,6 +116,9 @@ pytest test_calculator.py -v
 
 # Ejecutar el programa principal
 python main.py
+
+# Ejecutar la interfaz gráfica
+python gui.py
 ```
 
 ### Integración Continua (CI)

--- a/gui.py
+++ b/gui.py
@@ -165,11 +165,65 @@ class CalculatorGUI:
             >>> # 3. Ingresa "3"  
             >>> # 4. Click "=": calcula 5+3=8
         """
-        if self.current_value:
-            # Si hay operación pendiente, calcularla primero
-            if self.first_number is not None and self.operator:
-                self.equals_click()
-            
+        # Si el usuario presiona '-' cuando no hay número iniciado, iniciamos un número negativo
+        # Manejar '-' como inicio de número negativo
+        if operation == '-' and (self.current_value == "" or self.current_value is None):
+            # Iniciamos la entrada de un número negativo
+            self.current_value = '-'
+            self.display.delete(0, tk.END)
+            self.display.insert(0, self.current_value)
+            return
+        if operation == '-' and self.current_value == '-':
+            # Evitar varios signos '-'
+            return
+
+        # Si no hay nada en el display y ya hay un primer número,
+        # permitimos cambiar el operador (por ejemplo: 5 + -> 5 -)
+        if not self.current_value:
+            if self.first_number is not None:
+                self.operator = operation
+            return
+
+        # A partir de aquí current_value contiene texto: validar una vez
+        try:
+            value = float(self.current_value)
+        except ValueError:
+            self.show_error("Número inválido")
+            return
+
+        # Si no hay primer número guardado, lo asignamos
+        if self.first_number is None:
+            self.first_number = value
+            self.operator = operation
+            self.current_value = ""
+            return
+
+        # Si ya hay primer número y operador, calculamos la operación pendiente
+        if self.first_number is not None and self.operator is not None:
+            # Guardamos el segundo número y ejecutamos equals
+            self.equals_click()
+            # equals_click deja el resultado en self.current_value
+            try:
+                self.first_number = float(self.current_value)
+            except ValueError:
+                self.show_error("Número inválido")
+                return
+            self.operator = operation
+            self.current_value = ""
+
+        elif self.current_value and self.first_number is None and self.operator is None:
+            self.first_number = float(self.current_value)
+            self.operator = operation
+            self.current_value = ""
+
+        elif self.first_number is not None and self.operator is not None and self.current_value:
+            # validar current_value antes de llamar equals_click
+            try:
+                float(self.current_value)
+            except ValueError:
+                self.show_error("Número inválido")
+                return
+            self.equals_click()
             self.first_number = float(self.current_value)
             self.operator = operation
             self.current_value = ""
@@ -186,12 +240,12 @@ class CalculatorGUI:
             >>> # Ejecuta: add(5, 3) = 8
             >>> # Display: "8"
         """
-        if not self.current_value:
+        if self.first_number is not None and self.operator is not None and not self.current_value:
             # Muestra error en el display y resetea el estado
             self.show_error("Ingresa el segundo número")
             return
         
-        elif self.first_number is not None and self.operator:
+        elif self.first_number is not None and self.operator is not None and self.current_value:
             # En caso de no haber operador o primer número, no hacer nada
 
             try:
@@ -285,6 +339,9 @@ class CalculatorGUI:
 
     def unary_operation(self, func):
         if self.current_value:
+            if self.current_value == "-":
+                self.show_error("Número incompleto")
+                return
             try:
                 result = None
 

--- a/test_calculator.py
+++ b/test_calculator.py
@@ -14,12 +14,14 @@ def test_add():
     assert add(2, 3) == 5
     assert add(-1, 1) == 0
     assert add(0, 5) == 5
+    assert add(-2, -3) == -5
 
 
 def test_subtract():
     """Test resta básica."""
     assert subtract(5, 3) == 2
     assert subtract(0, 5) == -5
+    assert subtract(-2, -3) == 1
 
 
 def test_multiply():
@@ -27,13 +29,15 @@ def test_multiply():
     assert multiply(3, 4) == 12
     assert multiply(5, 0) == 0
     assert multiply(-2, 3) == -6
+    assert multiply(-2, -3) == 6
 
 
 def test_divide():
     """Test división básica."""
     assert divide(10, 2) == 5.0
     assert divide(7, 2) == 3.5
-
+    assert divide(-2, 3) == -0.6666666666666666
+    assert divide(-2, -3) == 0.6666666666666666
 
 def test_divide_by_zero():
     """Test que división por cero lanza excepción."""
@@ -46,6 +50,10 @@ def test_power():
     assert power(2, 3) == 8
     assert power(5, 0) == 1
     assert power(4, 0.5) == 2.0
+    assert power(-2, 3) == -8
+    assert power(-2, 0) == 1
+    assert power(-2, -3) == -0.125
+    # assert power(-4, 0.5) == -2.0, "Error: la raíz cuadrada de un número negativo no es real"
 
 
 def test_valor_maximo():

